src/targets/gpu/mlir.cpp

```diff
@@ -729,6 +729,7 @@ struct mlir_program
     void run_high_level_pipeline()
     {
         mlir_pass_manager pm_front{mlirPassManagerCreate(ctx.get())};
+        mlirRegisterRocMLIROptions();
```

Comment:
The calls have to go here because --mlir-print-ir-after-all, for instance, is a pass-manager option and can't be interpreted with the call-once MLIR registration functions.  That also means we can't hide the call in rocMLIR code.

Reply:
Also, why cant mlir just call this method in `mlirMIGraphXAddHighLevelPipeline` and `mlirMIGraphXAddBackendPipeline`?

Reply:
None of the calls from the registration function take the pass manager as input.  I think there's cl::opt magic that creates the options local to the pass manager and also enables parsing the command line to find them.

I'll try calling it in the other functions;  that would be a better choice.  I'll be back with the results.

Reply:
But do you need a mutex for that? This sounds like its not thread-safe.